pkg/cloudclient/cloudclient.go

```diff
@@ -25,22 +55,52 @@ type CloudClient interface {
 	// ValidateEgress validates that all required targets are reachable from the vpcsubnet
 	// target URLs: https://docs.openshift.com/rosa/rosa_getting_started/rosa-aws-prereqs.html#osd-aws-privatelink-firewall-prerequisites
 	// Expected return value is *output.Output that's storing failures, exceptions and errors
-	ValidateEgress(ctx context.Context, vpcSubnetID, cloudImageID string, kmsKeyID string, timeout time.Duration) *output.Output
+	ValidateEgress(ctx context.Context) *output.Output
 
 	// VerifyDns verifies that a given VPC meets the DNS requirements specified in:
 	// https://docs.openshift.com/container-platform/4.10/installing/installing_aws/installing-aws-vpc.html
 	// Expected return value is *output.Output that's storing failures, exceptions and errors
 	VerifyDns(ctx context.Context, vpcID string) *output.Output
 }
 
-func NewClient(ctx context.Context, logger ocmlog.Logger, creds interface{}, region, instanceType string, tags map[string]string) (CloudClient, error) {
-	switch c := creds.(type) {
-	case awscredsv1.Credentials, awscredsv2.StaticCredentialsProvider:
-		return awsCloudClient.NewClient(ctx, logger, c, region, instanceType, tags)
-	case *google.Credentials:
-		return gcpCloudClient.NewClient(ctx, logger, c, region, instanceType, tags)
+func getCloudClientType(options CmdOptions) string {
+	if options.AwsProfile != "" || os.Getenv("AWS_ACCESS_KEY_ID") != "" || options.CloudType == "aws" {
+		return AWS
+	}
+	if options.CloudType == GCP {
+		return GCP
+	}
+	return "unknown"
+}
+
+func NewClient(ctx context.Context, logger ocmlog.Logger,
+	options CmdOptions) (CloudClient, error) {
+	logger.Info(ctx, "Using region: %s", options.Region)
```

Comment:
This won't print out anything if profile isn't set on the cli, but the test still runs.

Reply:
Example:

```
 AWS_ACCESS_KEY_ID=$ACCESS_KEY AWS_SECRET_ACCESS_KEY=$SECRET_KEY ./osd-network-verifier egress --subnet-id $SUBNET_ID  
Using region: 
Using AWS secret key
Created instance with ID: i-06261950f8802ec25
EC2 Instance: i-06261950f8802ec25 Running
Gathering and parsing console log output...
Terminating ec2 instance with id i-06261950f8802ec25
Summary:
All tests pass!
Success
```

Reply:
I'd like to remove it